nemo/collections/nlp/models/text_classification_model.py

```diff
@@ -132,40 +172,44 @@ def validation_epoch_end(self, outputs):
         """
         avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
         # val_acc = sum([x['n_correct_pred'] for x in outputs]) / sum(x['n_pred'] for x in outputs)
-        # tensorboard_logs = {'val_loss': avg_loss, 'val_acc': val_acc}
-        return {'val_loss': avg_loss}  # , 'log': tensorboard_logs}
+        tensorboard_logs = {'val_loss': avg_loss}  # , 'val_acc': val_acc}
+        # return {'val_loss': avg_loss}  # , 'log': tensorboard_logs}
+        return {'val_loss': avg_loss, 'log': tensorboard_logs}
 
-    def setup_training_data(self, file_path, dataloader_params={}):
-        self._train_dl = self.__setup_dataloader(input_file=file_path, dataloader_params=dataloader_params)
+    def setup_training_data(self, train_data_config: Optional[DictConfig]):
+        # if 'shuffle' not in train_data_config:
+        #     train_data_config['shuffle'] = True
+        self._train_dl = self.__setup_dataloader_from_config(cfg=train_data_config)
 
-    def setup_validation_data(self, file_path, dataloader_params={}):
-        self._validation_dl = self.__setup_dataloader(input_file=file_path, dataloader_params=dataloader_params)
+    def setup_validation_data(self, val_data_config: Optional[DictConfig]):
+        self._validation_dl = self.__setup_dataloader_from_config(cfg=val_data_config)
 
-    def setup_test_data(self, file_path, dataloader_params={}):
-        self._test_dl = self.__setup_dataloader(input_file=file_path, dataloader_params=dataloader_params)
+    def setup_test_data(self, test_data_config: Optional[DictConfig]):
+        self._test_dl = self.__setup_dataloader(cfg=test_data_config)
 
-    def __setup_dataloader(
-        self, input_file, dataloader_params={},
-    ):
+    def __setup_dataloader_from_config(self, cfg: TextClassificationDataConfig):
```

Comment:
Do we need two underlines as the prefix?

Reply:
No, and we should strive not to use it. It messes up subclassing due to name mangling

Reply:
Happy to get rid of those extra "_"